plugins/node/opentelemetry-instrumentation-redis/src/instrumentation.ts

```diff
@@ -96,28 +108,116 @@ export class RedisInstrumentation extends InstrumentationBase<RedisInstrumentati
       ),
     ];
   }
+
   /**
    * Patch internal_send_command(...) to trace requests
    */
   private _getPatchInternalSendCommand() {
-    const tracer = this.tracer;
-    const config = this.getConfig();
+    const instrumentation = this;
     return function internal_send_command(original: Function) {
-      return getTracedInternalSendCommand(tracer, original, config);
+      return function internal_send_command_trace(
+        this: RedisPluginClientTypes,
+        cmd?: RedisCommand
+      ) {
+        // Versions of redis (2.4+) use a single options object
+        // instead of named arguments
+        if (arguments.length !== 1 || typeof cmd !== 'object') {
+          // We don't know how to trace this call, so don't start/stop a span
+          return original.apply(this, arguments);
+        }
+
+        const config = instrumentation.getConfig();
+
+        const hasNoParentSpan = trace.getSpan(context.active()) === undefined;
+        if (config.requireParentSpan === true && hasNoParentSpan) {
+          return original.apply(this, arguments);
+        }
+
+        const dbStatementSerializer =
+          config?.dbStatementSerializer || defaultDbStatementSerializer;
+        const span = instrumentation.tracer.startSpan(
```

Comment:
reviewer note: this is the fix. here we use `instrumentation.tracer` which gets updated in after `setTracerProvider` so we always use the latest one